Improve encoding section (#2735)
altair-viz/altair: doc/user_guide/encodings/index.rst

Here the y-axis is sorted reverse-alphabetically, while the color legend is
sorted in the specified order, beginning with ``'Morris'``.

Datum and Value
~~~~~~~~~~~~~~~

So far we always mapped an encoding channel to a column in our dataset. However, sometimes
it is also useful to map to a single constant value. In Altair, you can do this with

* ``datum``, which encodes a constant domain value via a scale using the same units as the underlying data
* ``value``, which encodes a constant visual value, using absolute units such as an exact position in pixels, the name or RGB value of a color, the name of shape,  etc

``datum`` is particularly useful for annotating a specific data value. 
For example, you can use it with a rule mark to highlight a 
threshold value (e.g., 300 dollars stock price).

.. altair-plot::
    
    import altair as alt
    from vega_datasets import data

    source = data.stocks()
    base = alt.Chart(source)
    lines = base.mark_line().encode(
        x="date:T",
        y="price:Q",
        color="symbol:N"
    )
    rule = base.mark_rule(strokeDash=[2, 2]).encode(
        y=alt.datum(300)
    )

    lines + rule

> If we instead used ``alt.value`` in this example, we would position the rule 300 pixels from the bottom of the chart border rather than at the 300 dollars position:

Reviewer: ```suggestion
If we instead used ``alt.value`` in this example, we would position the rule 300 pixels from the top of the chart border rather than at the 300 dollars position. Since the default charts height is 300 pixels, this will show the dotted line just on top of the x-axis -line:
```

.. altair-plot::

    rule = base.mark_rule(strokeDash=[2, 2]).encode(
        y=alt.value(300)
    )

    lines + rule

> You can also use ``datum`` with :class:`DateTime`, for example, to highlight a certain year.

Reviewer: ```suggestion
If we want to use ``datum``  to highlight a certain year on the x-axis,
we can't simply type in the year as an integer,
but instead need to use ``datum`` together with :class:`DateTime`.
```

> In addition, we will set the color for the rule to the same one as the line for the symbol ``MSFT``

Reviewer: ```suggestion
Here we also set the color for the rule to the same one as the line for the symbol ``MSFT``
```
